Add RPC timing in reco::Muon dataformat. Improve DT timing calculation

ptraczyk opened the pull request:
In order to allow out-of-time muon rejection during the muon ID stage of analyses two things have been changed:
1. hit pruning (outlier rejection) in DT has been switched on. The algorithm has been tweaked slightly.
2. RPC timing was added directly to the reco::Muon dataformat. The time calculation was added and timing code was re-organized in places.

The improvements in DT timing after the change have been shown here: https://indico.cern.ch/event/483080/contribution/1/attachments/1212148/1768319/ptraczyk_201601_muon_timing.pdf

Nothing besides the performance of DT timing (and CSC timing to a very small extent) should change.

ptraczyk committed:
switch on DT time measurement outlier rejection, clean up code, small update in outlier rejection logic

ptraczyk committed:
add the calculation of RPC time and store it directly in reco::Muon

ptraczyk committed:
add RPC timing to reco::Muon, small reorganization in the timing code

ptraczyk committed:
refactor timing code a little again

ptraczyk committed:
add fallback mechanism for old hlt configs

cmsbuild commented:
A new Pull Request was created by  @ptraczyk (Piotr Traczyk) for CMSSW_8_0_X.

It involves the following packages:

DataFormats/MuonReco
DataFormats/PatCandidates
RecoMuon/Configuration
RecoMuon/MuonIdentification
RecoMuon/TrackingTools


@cvuosalo, @monttj, @cmsbuild, @slava77, @vadler, @davidlange6 can you please review it and eventually sign? Thanks.
@battibass, @abbiendi, @jhgoh, @bellan, @gpetruc, @trocino, @amagitte, @bachtis, @rociovilar this is something you requested to watch as well.
@slava77, @degano, @smuzaffar you are the release manager for this.

Following commands in first line of a comment are recognized

* **+1|approve[d]|sign[ed]**: L1/L2's to approve it
* **-1|reject[ed]**: L1/L2's to reject it
* **assign &lt;category&gt;[,&lt;category&gt;[,...]]**: L1/L2's to request signatures from other categories
* **unassign &lt;category&gt;[,&lt;category&gt;[,...]]**: L1/L2's to remove signatures from other categories
* **hold**: L1/all L2's/release manager to mark it as on hold
* **unhold**: L1/user who put this PR on hold
* **merge**: L1/release managers to merge this request
* **[@cmsbuild,] please test**: L1/L2 and selected users to start jenkins tests
* **[@cmsbuild,] please test with cms-sw/cmsdist#&lt;PR&gt;**: L1/L2 and selected users to start jenkins tests using externals from cmsdist

slava77 commented on RecoMuon/MuonIdentification/src/MuonTimingFiller.cc:
```diff
@@ -44,11 +45,21 @@ MuonTimingFiller::MuonTimingFiller(const edm::ParameterSet& iConfig, edm::Consum
+       matchParameters = dtTimingParameters.getParameter<edm::ParameterSet>("MatchParameters");
+   
+   
+   theMatcher_ = new MuonSegmentMatcher(matchParameters, iC);
```
it could be a good opportunity to switch to unique_ptr to let it handle the lifetime of the object and drop the delete

ptraczyk replied:
On Thu, Jan 21, 2016 at 7:00 AM, Slava Krutelyov <notifications@github.com>
wrote:

> In RecoMuon/MuonIdentification/src/MuonTimingFiller.cc
> <https://github.com/cms-sw/cmssw/pull/13010#discussion_r50363402>:
>
> >
> >     // Load parameters for the CSCTimingExtractor
> >     edm::ParameterSet cscTimingParameters = iConfig.getParameter<edm::ParameterSet>("CSCTimingParameters");
> > -   theCSCTimingExtractor_ = new CSCTimingExtractor(cscTimingParameters,iC);
> > +
> > +   // Fallback mechanism for old configs (there the segment matcher was built inside the timing extractors)
> > +   edm::ParameterSet matchParameters;
> > +   if (iConfig.existsAs<edm::ParameterSet>("MatchParameters"))
> > +       matchParameters = iConfig.getParameter<edm::ParameterSet>("MatchParameters");
> > +     else
> > +       matchParameters = dtTimingParameters.getParameter<edm::ParameterSet>("MatchParameters");
> > +
> > +
> > +   theMatcher_ = new MuonSegmentMatcher(matchParameters, iC);
>
> it could be a good opportunity to switch to unique_ptr to let it handle
> the lifetime of the object and drop the delete
>
I'll do that, tomorrow if I find the time. I guess I can make another PR
for that once this one goes in.



> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/13010/files#r50363402>.
>

slava77 replied:
updates in a followup PR should be OK

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 6509b8bd8bed80400ce406a4427c5f8fad45df24
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13010/10654/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13010/10654/summary.html

slava77 commented:
In the 10 GeV muon gun:
- regular muons (the same 10 GeV gun) time reco is not affected significantly. There is a component with reduced uncertainty values (from the DT changes?)
![all_sign662vsorig_singlemupt10wf20p0c_recomuons_muons__reco_obj_time_timeatipinout](https://cloud.githubusercontent.com/assets/4676718/12514386/f9d35e72-c0d7-11e5-87b3-2fddc9062a8d.png)
![all_sign662vsorig_singlemupt10wf20p0c_recomuons_muons__reco_obj_time_timeatipinouterr](https://cloud.githubusercontent.com/assets/4676718/12514413/1689e680-c0d8-11e5-8844-8b8398b4dc38.png)
- less measurements contribute to 1-leg cosmic reco. I suppose, this is expected, since these are not real 1-leg cosmic muons and the introduced DT pruning is expected to drop some hits. The mean time is more broad but is also more centered around 0.
![all_sign662vsorig_singlemupt10wf20p0c_recomuons_muonsfromcosmics1leg__reco_obj_time_timeatipinouterr](https://cloud.githubusercontent.com/assets/4676718/12514287/7c98d658-c0d7-11e5-8de7-64cec8b4bf5f.png)
![all_sign662vsorig_singlemupt10wf20p0c_recomuons_muonsfromcosmics1leg__reco_obj_time_ndof](https://cloud.githubusercontent.com/assets/4676718/12514257/4c7aec0e-c0d7-11e5-90da-62eb0245165f.png)
![all_sign662vsorig_singlemupt10wf20p0c_recomuons_muonsfromcosmics1leg__reco_obj_time_timeatipinout](https://cloud.githubusercontent.com/assets/4676718/12514269/5de06622-c0d7-11e5-8e8b-27d0406c6aa8.png)
- the "regular" cosmic reco in the 10 GeV gun is changed less in terms of the number of measurements, but the in->out time now has a tail to more positive values. The part around/above zero is supposed to correspond to muons at the bottom and I would expect they should look like prompt and have no extra positive time tail.
![all_sign662vsorig_singlemupt10wf20p0c_recomuons_muonsfromcosmics__reco_obj_time_ndof](https://cloud.githubusercontent.com/assets/4676718/12514323/a9d22e58-c0d7-11e5-8a83-e93d2d965f3c.png)
![all_sign662vsorig_singlemupt10wf20p0c_recomuons_muonsfromcosmics__reco_obj_time_timeatipinout](https://cloud.githubusercontent.com/assets/4676718/12514349/d48e6800-c0d7-11e5-8595-26000c3fb80a.png)

@ptraczyk 
Are the changes in cosmicMuons in this gun sample (the previous plot) OK?

ptraczyk commented:
ok, this is something that I overlooked. For the outside-in cosmic tracks
this will not be working ok since hit pruning is working taking the mean
hit time as the reference for outlier rejection, and this time is
calculated with the inside-out assumption. So the mechanism should be
turned off for the cosmic muon collections. This should be possible, I'll
look into the configs now.

On Fri, Jan 22, 2016 at 4:34 PM, Slava Krutelyov <notifications@github.com>
wrote:

> In the 10 GeV muon gun:
>
>    - regular muons (the same 10 GeV gun) time reco is not affected
>    significantly. There is a component with reduced uncertainty values (from
>    the DT changes?) [image:
>    all_sign662vsorig_singlemupt10wf20p0c_recomuons_muons__reco_obj_time_timeatipinout]
>    <https://cloud.githubusercontent.com/assets/4676718/12514386/f9d35e72-c0d7-11e5-87b3-2fddc9062a8d.png> [image:
>    all_sign662vsorig_singlemupt10wf20p0c_recomuons_muons__reco_obj_time_timeatipinouterr]
>    <https://cloud.githubusercontent.com/assets/4676718/12514413/1689e680-c0d8-11e5-8844-8b8398b4dc38.png>
>    - less measurements contribute to 1-leg cosmic reco. I suppose, this
>    is expected, since these are not real 1-leg cosmic muons and the introduced
>    DT pruning is expected to drop some hits. The mean time is more broad but
>    is also more centered around 0. [image:
>    all_sign662vsorig_singlemupt10wf20p0c_recomuons_muonsfromcosmics1leg__reco_obj_time_timeatipinouterr]
>    <https://cloud.githubusercontent.com/assets/4676718/12514287/7c98d658-c0d7-11e5-8de7-64cec8b4bf5f.png> [image:
>    all_sign662vsorig_singlemupt10wf20p0c_recomuons_muonsfromcosmics1leg__reco_obj_time_ndof]
>    <https://cloud.githubusercontent.com/assets/4676718/12514257/4c7aec0e-c0d7-11e5-90da-62eb0245165f.png> [image:
>    all_sign662vsorig_singlemupt10wf20p0c_recomuons_muonsfromcosmics1leg__reco_obj_time_timeatipinout]
>    <https://cloud.githubusercontent.com/assets/4676718/12514269/5de06622-c0d7-11e5-8e8b-27d0406c6aa8.png>
>    - the "regular" cosmic reco in the 10 GeV gun is changed less in terms
>    of the number of measurements, but the in->out time now has a tail to more
>    positive values. The part around/above zero is supposed to correspond to
>    muons at the bottom and I would expect they should look like prompt and
>    have no extra positive time tail. [image:
>    all_sign662vsorig_singlemupt10wf20p0c_recomuons_muonsfromcosmics__reco_obj_time_ndof]
>    <https://cloud.githubusercontent.com/assets/4676718/12514323/a9d22e58-c0d7-11e5-8a83-e93d2d965f3c.png> [image:
>    all_sign662vsorig_singlemupt10wf20p0c_recomuons_muonsfromcosmics__reco_obj_time_timeatipinout]
>    <https://cloud.githubusercontent.com/assets/4676718/12514349/d48e6800-c0d7-11e5-8595-26000c3fb80a.png>
>
> @ptraczyk <https://github.com/ptraczyk>
> Are the changes in cosmicMuons in this gun sample (the previous plot) OK?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/13010#issuecomment-173952842>.
>

ptraczyk commented:
oh, and concerning the 10 GeV particle gun - the number of measurements
should go down a little, but the cleanup maily affects the left-side tail
of the time measurement, which is not big in the first place. So on a small
particle gun sample the improvement is not obvious.

On Fri, Jan 22, 2016 at 5:08 PM, Piotr Traczyk <piotraczyk@gmail.com> wrote:

> ok, this is something that I overlooked. For the outside-in cosmic tracks
> this will not be working ok since hit pruning is working taking the mean
> hit time as the reference for outlier rejection, and this time is
> calculated with the inside-out assumption. So the mechanism should be
> turned off for the cosmic muon collections. This should be possible, I'll
> look into the configs now.
>
> On Fri, Jan 22, 2016 at 4:34 PM, Slava Krutelyov <notifications@github.com
> > wrote:
>
>> In the 10 GeV muon gun:
>>
>>    - regular muons (the same 10 GeV gun) time reco is not affected
>>    significantly. There is a component with reduced uncertainty values (from
>>    the DT changes?) [image:
>>    all_sign662vsorig_singlemupt10wf20p0c_recomuons_muons__reco_obj_time_timeatipinout]
>>    <https://cloud.githubusercontent.com/assets/4676718/12514386/f9d35e72-c0d7-11e5-87b3-2fddc9062a8d.png> [image:
>>    all_sign662vsorig_singlemupt10wf20p0c_recomuons_muons__reco_obj_time_timeatipinouterr]
>>    <https://cloud.githubusercontent.com/assets/4676718/12514413/1689e680-c0d8-11e5-8844-8b8398b4dc38.png>
>>    - less measurements contribute to 1-leg cosmic reco. I suppose, this
>>    is expected, since these are not real 1-leg cosmic muons and the introduced
>>    DT pruning is expected to drop some hits. The mean time is more broad but
>>    is also more centered around 0. [image:
>>    all_sign662vsorig_singlemupt10wf20p0c_recomuons_muonsfromcosmics1leg__reco_obj_time_timeatipinouterr]
>>    <https://cloud.githubusercontent.com/assets/4676718/12514287/7c98d658-c0d7-11e5-8de7-64cec8b4bf5f.png> [image:
>>    all_sign662vsorig_singlemupt10wf20p0c_recomuons_muonsfromcosmics1leg__reco_obj_time_ndof]
>>    <https://cloud.githubusercontent.com/assets/4676718/12514257/4c7aec0e-c0d7-11e5-90da-62eb0245165f.png> [image:
>>    all_sign662vsorig_singlemupt10wf20p0c_recomuons_muonsfromcosmics1leg__reco_obj_time_timeatipinout]
>>    <https://cloud.githubusercontent.com/assets/4676718/12514269/5de06622-c0d7-11e5-8e8b-27d0406c6aa8.png>
>>    - the "regular" cosmic reco in the 10 GeV gun is changed less in
>>    terms of the number of measurements, but the in->out time now has a tail to
>>    more positive values. The part around/above zero is supposed to correspond
>>    to muons at the bottom and I would expect they should look like prompt and
>>    have no extra positive time tail. [image:
>>    all_sign662vsorig_singlemupt10wf20p0c_recomuons_muonsfromcosmics__reco_obj_time_ndof]
>>    <https://cloud.githubusercontent.com/assets/4676718/12514323/a9d22e58-c0d7-11e5-8a83-e93d2d965f3c.png> [image:
>>    all_sign662vsorig_singlemupt10wf20p0c_recomuons_muonsfromcosmics__reco_obj_time_timeatipinout]
>>    <https://cloud.githubusercontent.com/assets/4676718/12514349/d48e6800-c0d7-11e5-8595-26000c3fb80a.png>
>>
>> @ptraczyk <https://github.com/ptraczyk>
>> Are the changes in cosmicMuons in this gun sample (the previous plot) OK?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/cms-sw/cmssw/pull/13010#issuecomment-173952842>.
>>
>
>

ptraczyk committed:
switch off hit pruning for cosmics

cmsbuild commented:
Pull request #13010 was updated. @cvuosalo, @monttj, @cmsbuild, @slava77, @vadler, @davidlange6 can you please check and sign again.

ptraczyk commented:
I switched OFF hit pruning in the cosmic configs. In general it should be
switched off by the user in all non-standard cases (HSCP, cosmics etc). It
is ON in the default measurement to allow for the best performing rejection
of muons from other bunch crossings for all analyses.

On Fri, Jan 22, 2016 at 8:19 PM, cmsbuild <notifications@github.com> wrote:

> Pull request #13010 <https://github.com/cms-sw/cmssw/pull/13010> was
> updated. @cvuosalo <https://github.com/cvuosalo>, @monttj
> <https://github.com/monttj>, @cmsbuild <https://github.com/cmsbuild>,
> @slava77 <https://github.com/slava77>, @vadler <https://github.com/vadler>,
> @davidlange6 <https://github.com/davidlange6> can you please check and
> sign again.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/13010#issuecomment-174016974>.
>

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 18620b72f88b03efcc48f56c935d074904106309
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13010/10686/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13010/10686/summary.html

slava77 commented:
After the last update the number of hits is roughly the same in cosmics (the pruning is not applied)
In the same 10 GeV muon gun now:
![all_sign662-pass-18620b7vsorig_singlemupt10wf20p0c_recomuons_muonsfromcosmics1leg__reco_obj_time_ndof](https://cloud.githubusercontent.com/assets/4676718/12532826/45ce1736-c1d2-11e5-8512-4a25c077fecc.png)
![all_sign662-pass-18620b7vsorig_singlemupt10wf20p0c_recomuons_muonsfromcosmics1leg__reco_obj_time_timeatipinout](https://cloud.githubusercontent.com/assets/4676718/12532828/4b4bd720-c1d2-11e5-8c8f-e4ebb1fe7e93.png)
![all_sign662-pass-18620b7vsorig_singlemupt10wf20p0c_recomuons_muonsfromcosmics1leg__reco_obj_time_timeatipinouterr](https://cloud.githubusercontent.com/assets/4676718/12532830/7041c490-c1d2-11e5-82f3-782b5f99ba20.png)

and for separately reconstructed cosmics the prompt-like direction is a bit more centered at 0; but I do not understand the growing peak at ~15ns (is there something about dt4DCosmicSegments that leads to this discrepancy?)
![all_sign662-pass-18620b7vsorig_singlemupt10wf20p0c_recomuons_muonsfromcosmics__reco_obj_time_timeatipinout](https://cloud.githubusercontent.com/assets/4676718/12532845/b307586c-c1d2-11e5-8a43-70f2c4ef3166.png)
Here is the eta-phi distribution for muons with new time>10 and |time_new - time_old| > 10. The majority here has the same nDof before and after in the range of ~25 and above
![wf20 0_cosmulargedt_eta_phi](https://cloud.githubusercontent.com/assets/4676718/12532968/2623fe56-c1d6-11e5-8e7d-fe42d1ec745d.png)

ptraczyk commented:
I don't understand the small peak at 15ns in cosmics. I suspect it can be
due to a small change that I made to the logic determining propagation
direction based of inner and outer position .. maybe it's getting confused
for 1-leg cosmics.
Investigating.

On Sat, Jan 23, 2016 at 10:52 PM, Slava Krutelyov <notifications@github.com>
wrote:

> After the last update the number of hits is roughly the same in cosmics
> (the pruning is not applied)
> In the same 10 GeV muon gun now:
> [image:
> all_sign662-pass-18620b7vsorig_singlemupt10wf20p0c_recomuons_muonsfromcosmics1leg__reco_obj_time_ndof]
> <https://cloud.githubusercontent.com/assets/4676718/12532826/45ce1736-c1d2-11e5-8512-4a25c077fecc.png>
> [image:
> all_sign662-pass-18620b7vsorig_singlemupt10wf20p0c_recomuons_muonsfromcosmics1leg__reco_obj_time_timeatipinout]
> <https://cloud.githubusercontent.com/assets/4676718/12532828/4b4bd720-c1d2-11e5-8c8f-e4ebb1fe7e93.png>
> [image:
> all_sign662-pass-18620b7vsorig_singlemupt10wf20p0c_recomuons_muonsfromcosmics1leg__reco_obj_time_timeatipinouterr]
> <https://cloud.githubusercontent.com/assets/4676718/12532830/7041c490-c1d2-11e5-82f3-782b5f99ba20.png>
>
> and for separately reconstructed cosmics the prompt-like direction is a
> bit more centered at 0; but I do not understand the growing peak at ~15ns
> (is there something about dt4DCosmicSegments that leads to this
> discrepancy?)
> [image:
> all_sign662-pass-18620b7vsorig_singlemupt10wf20p0c_recomuons_muonsfromcosmics__reco_obj_time_timeatipinout]
> <https://cloud.githubusercontent.com/assets/4676718/12532845/b307586c-c1d2-11e5-8a43-70f2c4ef3166.png>
> Here is the eta-phi distribution for muons with new time>10 and |time_new
> - time_old| > 10. The majority here has the same nDof before and after in
> the range of ~25 and above
> [image: wf20 0_cosmulargedt_eta_phi]
> <https://cloud.githubusercontent.com/assets/4676718/12532968/2623fe56-c1d6-11e5-8e7d-fe42d1ec745d.png>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/13010#issuecomment-174225271>.
>

ptraczyk commented:
which sample were these plots made on? I'm trying to reproduce this..

On Mon, Jan 25, 2016 at 11:29 AM, Piotr Traczyk <piotraczyk@gmail.com>
wrote:

> I don't understand the small peak at 15ns in cosmics. I suspect it can be
> due to a small change that I made to the logic determining propagation
> direction based of inner and outer position .. maybe it's getting confused
> for 1-leg cosmics.
> Investigating.
>
> On Sat, Jan 23, 2016 at 10:52 PM, Slava Krutelyov <
> notifications@github.com> wrote:
>
>> After the last update the number of hits is roughly the same in cosmics
>> (the pruning is not applied)
>> In the same 10 GeV muon gun now:
>> [image:
>> all_sign662-pass-18620b7vsorig_singlemupt10wf20p0c_recomuons_muonsfromcosmics1leg__reco_obj_time_ndof]
>> <https://cloud.githubusercontent.com/assets/4676718/12532826/45ce1736-c1d2-11e5-8512-4a25c077fecc.png>
>> [image:
>> all_sign662-pass-18620b7vsorig_singlemupt10wf20p0c_recomuons_muonsfromcosmics1leg__reco_obj_time_timeatipinout]
>> <https://cloud.githubusercontent.com/assets/4676718/12532828/4b4bd720-c1d2-11e5-8c8f-e4ebb1fe7e93.png>
>> [image:
>> all_sign662-pass-18620b7vsorig_singlemupt10wf20p0c_recomuons_muonsfromcosmics1leg__reco_obj_time_timeatipinouterr]
>> <https://cloud.githubusercontent.com/assets/4676718/12532830/7041c490-c1d2-11e5-82f3-782b5f99ba20.png>
>>
>> and for separately reconstructed cosmics the prompt-like direction is a
>> bit more centered at 0; but I do not understand the growing peak at ~15ns
>> (is there something about dt4DCosmicSegments that leads to this
>> discrepancy?)
>> [image:
>> all_sign662-pass-18620b7vsorig_singlemupt10wf20p0c_recomuons_muonsfromcosmics__reco_obj_time_timeatipinout]
>> <https://cloud.githubusercontent.com/assets/4676718/12532845/b307586c-c1d2-11e5-8a43-70f2c4ef3166.png>
>> Here is the eta-phi distribution for muons with new time>10 and |time_new
>> - time_old| > 10. The majority here has the same nDof before and after in
>> the range of ~25 and above
>> [image: wf20 0_cosmulargedt_eta_phi]
>> <https://cloud.githubusercontent.com/assets/4676718/12532968/2623fe56-c1d6-11e5-8e7d-fe42d1ec745d.png>
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/cms-sw/cmssw/pull/13010#issuecomment-174225271>.
>>
>
>

slava77 commented:
On 1/25/16 3:13 AM, Piotr Traczyk wrote:
> which sample were these plots made on? I'm trying to reproduce this..

This was matrix workflow 20.0
runTheMatrix.py -i all -l 20.0 --command="-n 1000"

ptraczyk commented:
Thanks.
It was the inner vs outer position thing that got confused by the cosmic
tracks. I removed it and it's now ok. I'll send a commit in a moment, I'm
also putting
std::unique_ptr in MuonTimingFiller.cc to have that also done

On Mon, Jan 25, 2016 at 2:15 PM, Slava Krutelyov <notifications@github.com>
wrote:

> On 1/25/16 3:13 AM, Piotr Traczyk wrote:
> > which sample were these plots made on? I'm trying to reproduce this..
>
> This was matrix workflow 20.0
> runTheMatrix.py -i all -l 20.0 --command="-n 1000"
>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/13010#issuecomment-174505391>.
>

ptraczyk committed:
remove DT propagation direction detection (was failing for cosmics), move to unique_ptr in timing code

cmsbuild commented:
Pull request #13010 was updated. @cvuosalo, @monttj, @cmsbuild, @slava77, @vadler, @davidlange6 can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1 
 
Tested at: 6ead007c877ff6d784ffc90aafeeab24941c0851
I found errors in the following addon tests: 
 
 cmsDriver.py RelVal -s L1REPACK:GT2 --data --scenario=HeavyIons -n 10 --conditions auto:run2_hlt_HIon --relval 9000,50 --datatier "RAW" --eventcontent RAW --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_HI --magField 38T_PostLS1 --fileout file:RelVal_Raw_HIon_DATA.root --filein /store/hidata/HIRun2015/HIHardProbes/RAW-RECO/HighPtJet-PromptReco-v1/000/263/689/00000/1802CD9A-DDB8-E511-9CF9-02163E0138CA.root : FAILED - time: date Tue Jan 26 22:05:47 2016-date Tue Jan 26 22:05:05 2016 s - exit: 23552
cmsRun /afs/cern.ch/cms/sw/ReleaseCandidates/vol1/slc6_amd64_gcc493/cms/cmssw-patch/CMSSW_8_0_X_2016-01-25-2300/src/HLTrigger/Configuration/test/OnLine_HLT_HIon.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Tue Jan 26 22:05:47 2016-date Tue Jan 26 22:05:05 2016 s - exit: 21504
cmsDriver.py RelVal -s HLT:HIon,RAW2DIGI,L1Reco,RECO --data --scenario=HeavyIons -n 10 --conditions auto:run2_data_HIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_HI --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_HIon_DATA.root --fileout file:RelVal_Raw_HIon_DATA_HLT_RECO.root : FAILED - time: date Tue Jan 26 22:05:47 2016-date Tue Jan 26 22:05:05 2016 s - exit: 21504
 
 
you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13010/10749/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13010/10749/summary.html

slava77 commented:
looking at https://github.com/cms-sw/cmssw/pull/13010	6ead007 in CMSSW_8_0_0_pre4:
- time for muonsFromCosmics in prompt samples now looks more like expected: the downward going muons (correct timing) now have a better centere peak at zero and no extra unwanted features
e.g. 10 GeV muon gun
![all_sign662-pass-6ead007vsorig_singlemupt10wf20p0c_recomuons_muonsfromcosmics__reco_obj_time_timeatipinout](https://cloud.githubusercontent.com/assets/4676718/12603656/7a19dc02-c469-11e5-8738-1cae137e16bd.png)

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/13010	6ead007
- code changes are in line with the PR description
- jenkins tests pass and comparisons with baseline show differences only in muon time distributions (small differences given the statistics of the tested samples)
- local tests with more samples and more events are in line with expectations for this update, some plots are posted earlier in this thread.

slava77 commented:
... the failure in addOn tests can be ignored, it is due to a missing input file, unrelated to this PR
